src/oc_erchef/apps/oc_chef_wm/src/oc_chef_action_insights.erl

```diff
@@ -0,0 +1,67 @@
+%% Copyright 2012-2014 Chef Software, Inc. All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+
+%% @doc Receiver of ingest requests. Requests are queued in a file and parsed
+%% asynchronously.
+
+-module(oc_chef_action_insights).
+
+-behaviour(gen_server).
+
+%% API
+-export([
+         ingest/2,
+         start_link/0
+        ]).
+
+%% gen_server callbacks
+-export([
+         init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3
+        ]).
+
+-spec ingest(ShouldSend :: boolean(), Data :: binary()) -> ok.
+ingest(true, Data) ->
+    gen_server:cast(?MODULE, {ingest, Data});
+ingest(false, _) -> ok.
+
+-spec start_link() -> {ok, pid()}.
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+init([]) ->
+    file:open(envy:get(oc_chef_wm, insights_log, string), [append]).
+
+handle_call(_, _, State) ->
+    {reply, ok , State}.
+
+handle_cast({ingest, Data}, IODevice) ->
+    file:write(IODevice, [Data, "\n"]),
```

Comment:
More of a note to self: I'm assuming that we are writing the newline since the ingester is going to be line-oriented?  I believe that our json encoding library ensures that any newlines are transformed escaped so I think we are safe from there being any newlines in the data.  If that isn't the case, an alternative might be to write a fix-length field that would store the length of the content and then write the message.  The ingester could then read the length and then read the specified number of bytes.